### PR TITLE
New mvaTTH weights for Run 3

### PR DESCRIPTION
**Pull request by Cvico**
## Brief description
Dear reviewers,

This PR is to introduce the newest training of the Electron mvaTTH in CMSSW. The training has been performed following what was done already done for Run 2, but using Run 3 MC samples for training and validation.

## Context of the PR
The goal for this new training is to be used as a substitute of the current mvaTTH **only in Run 3 MC production**. That is: for Run 2 we still need to keep the current mvaTTH implementation.

## What this PR includes
**This change should only affect Run 3 production**.  How the mva is evaluated changes a little bit with respect to the previous version: mainly the order in which the variables are parsed to TMVA and also we have added a new additional variable which is `LepGood_pfRelIso03_all`. 

## IMPORTANT REMARK
The training has been done considering a MiniAOD that has a buggy mvaNoISO variable. Instead of considering the buggy variable, the training has been done with mvaISO (and so it reflects in the config file, the last input variable to the MVA is mvaISO). This will change in the future most likely.

## Validation
I've performed a local test running a simple `cmsRun` script to see that the mva is loaded and nanoAOD is in fact produced. Below is  the cmsDriver I've used (in a cmssw-el8 container):

`cmsDriver.py  --python_filename GEN-Run3Summer23NanoAODv12-00146_1_cfg.py --eventcontent NANOEDMAODSIM --customise Configuration/DataProcessing/Utils.addMonitoring --datatier NANOAODSIM --fileout file:GEN-Run3Summer23NanoAODv12-00146.root --conditions 130X_mcRun3_2023_realistic_v14 --step NANO --scenario pp --filein "dbs:/WZto3LNu_TuneCP5_13p6TeV_powheg-pythia8/Run3Summer23MiniAODv4-130X_mcRun3_2023_realistic_v14-v1/MINIAODSIM" --era Run3_2023 --no_exec --mc -n 1
`
Below  there's also a plot comparing the performance of this training vs the current one.
image

## Other comments
Please let me know if the proposal is reasonable, and if there's anything else I can do for testing. I'm adding the L2 electron conveners @RSalvatico @afiqaize so they can follow up the PR as well. Feel free to add any other electron convener. I'm also tagging the placeholder PR that Riccardo prepare in the links below (I prefered creating a new one).

## Linked PRs
- This is the link for the PR that includes the weights in CMS-data: [pull19](https://github.com/cms-data/PhysicsTools-NanoAOD/pull/19).
- Placeholder PR: https://github.com/cms-sw/cmssw/pull/46316

### CODE REVIEW

**Comment by RSalvatico** [comment]
enable nano

**Comment by RSalvatico** [comment]
@cmsbuild, please test

**Comment by hqucms** [comment]
please abort

**Comment by hqucms** [comment]
test parameters:
- pull_request = https://github.com/cms-data/PhysicsTools-NanoAOD/pull/19

File: PhysicsTools/NanoAOD/python/electrons_cff.py [hunk 1]
```diff
@@ -283,11 +300,18 @@ def _get_bitmapVIDForEle_docstring(modules,WorkingPoints):
...
-    )
 )
+
 run2_egamma_2016.toModify(
     electronPROMPTMVA,
     weightFile = "PhysicsTools/NanoAOD/data/el_BDTG_2016.weights.xml",
+    variables = _legacy_electron_BDT_variable
+)
+
+(run2_egamma_2016 | run2_egamma_2018).toModify(
```
Comment (hqucms) [inline PhysicsTools/NanoAOD/python/electrons_cff.py:311]: @Cvico I think it should be `run2_egamma_2017` instead of `run2_egamma_2016`?

Comment (Cvico) [reply PhysicsTools/NanoAOD/python/electrons_cff.py]: Hi @hqucms. You are right, I've just pushed the fix. Thanks!

**Comment by hqucms** [comment]
please test

**Comment by Cvico** [comment]
Hi, sorry for delayed action, I'm at a conference. I think the issue above was due to a wrong call to pfRelIso03 variable (which is new in the training). I've taken the correct definition from the config file.

**Comment by hqucms** [comment]
please test

**Comment by hqucms** [comment]
+1

**Comment by RSalvatico** [comment]
@cms-sw/orp-l2 kind reminder of this and of the related PR to `cms-data`. We would like to meet tomorrow's deadline for integration in Nano v15. 

Thanks

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**